File: Monica for Android/app/src/main/java/takagi/ru/monica/data/PasswordEntry.kt
package takagi.ru.monica.data

import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import kotlinx.parcelize.Parcelize
import java.util.Date

/**
 * Password entry entity for Room database
 */
@Parcelize
@Entity(
    tableName = "password_entries",
    indices = [Index(value = ["isDeleted"])]
)
data class PasswordEntry(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val title: String,
    val website: String,
    val username: String,
    val password: String, // This will be encrypted
    val notes: String = "",
    val createdAt: Date = Date(),
    val updatedAt: Date = Date(),
    val isFavorite: Boolean = false,
    val sortOrder: Int = 0, // 排序顺序(用于拖动排序)
    val isGroupCover: Boolean = false, // 是否作为分组封面
    val appPackageName: String = "", // 关联的应用包名（用于自动填充匹配）
    val appName: String = "", // 关联的应用名称（用于显示）
    
    // Phase 7: 个人信息字段
    val email: String = "",
    val phone: String = "",
    
    // Phase 7: 地址信息字段
    val addressLine: String = "",
    val city: String = "",
    val state: String = "",
    val zipCode: String = "",
    val country: String = "",
    
    // Phase 7: 支付信息字段 (加密存储)
    val creditCardNumber: String = "",      // 加密存储
    val creditCardHolder: String = "",
    val creditCardExpiry: String = "",       // 格式: MM/YY
    val creditCardCVV: String = "",           // 加密存储
    
    val categoryId: Long? = null, // 分类ID
    
    // 本地 KeePass 数据库归属
    @ColumnInfo(defaultValue = "NULL")
    val keepassDatabaseId: Long? = null, // 归属的 KeePass 数据库ID
    
    // 关联的验证器密钥 (TOTP Secret)
    val authenticatorKey: String = "",  // 用于存储绑定的TOTP验证器密钥
    
    // 第三方登录(SSO)字段
    @ColumnInfo(defaultValue = "PASSWORD")
    val loginType: String = "PASSWORD",  // 登录类型: PASSWORD 或 SSO
    @ColumnInfo(defaultValue = "")
    val ssoProvider: String = "",        // SSO提供商: GOOGLE, APPLE, FACEBOOK 等
    @ColumnInfo(defaultValue = "NULL")
    val ssoRefEntryId: Long? = null,     // 引用的账号条目ID
    
    // 回收站功能 - 软删除字段
    @ColumnInfo(defaultValue = "0")
    val isDeleted: Boolean = false,      // 是否已删除（在回收站中）
    @ColumnInfo(defaultValue = "NULL")
    val deletedAt: java.util.Date? = null // 删除时间（用于自动清空）
) : Parcelable {
    
    /**
     * 是否使用第三方登录
     */
    fun isSsoLogin(): Boolean = loginType == "SSO"
    
    /**
     * 获取SSO提供商枚举
     */
    fun getSsoProviderEnum(): SsoProvider? {
        return if (isSsoLogin() && ssoProvider.isNotEmpty()) {
            SsoProvider.fromName(ssoProvider)
        } else null
    }
}

File: Monica for Windows/Monica.Windows/Models/PasswordEntry.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Monica.Windows.Models
{
    [Table("PasswordEntries")]
    public class PasswordEntry
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Required]
        public string Title { get; set; } = string.Empty;

        [NotMapped]
        public string TitleInitial => !string.IsNullOrEmpty(Title) ? Title.Substring(0, 1).ToUpper() : "?";

        public string Website { get; set; } = string.Empty;

        public string Username { get; set; } = string.Empty;

        [Required]
        public string EncryptedPassword { get; set; } = string.Empty; // Encrypted

        public string Notes { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
        public bool IsFavorite { get; set; }
        public int SortOrder { get; set; }

        public string? Email { get; set; }
        public string? Phone { get; set; }

        // Address
        public string? AddressLine { get; set; }
        public string? City { get; set; }
        public string? State { get; set; }
        public string? ZipCode { get; set; }
        public string? Country { get; set; }

        // Payment (Encrypted)
        public string? EncryptedCreditCardNumber { get; set; }
        public string? CreditCardHolder { get; set; }
        public string? CreditCardExpiry { get; set; } // MM/YY
        public string? EncryptedCreditCardCVV { get; set; }

        public int? CategoryId { get; set; }
        
        [ForeignKey("CategoryId")]
        public virtual Category? Category { get; set; }
    }
}
